lib/seattleflu/id3c/cli/command/clinical.py

```diff
@@ -291,6 +292,15 @@ def parse_sch(sch_filename, output):
 
 
 
+
+def drop_unnecessary_columns(df: pd.DataFrame, columns_to_keep = List[str]) -> pd.DataFrame:
+    """
+    Drops all columns from a given *df* except those explicitly listed in *columns_to_keep*. Returns
+    the new DataFrame.
+    """
+    return df[columns_to_keep]
```

Comment:
On the face of it, this seems like refactoring gone too far. What's the motivation here? (The commit message doesn't say why, just what. :-)

Reply:
That's fair. I thought it made the code more readable. By refactoring it and naming the function after what it does, I removed a comment that said "Drop unnecessary columns".
If you think that's maybe overkill then I understand.

Reply:
Nod. I think it's skewing a little too far. Indexing/subsetting operations like `df[columns]` are a pretty fundamental syntax. Wrapping that up into a function means more cognitive overhead (for me) because I have to recall if the function is just `df[columns]` or something more involved (with the latter being my assumed default since it's a function). It also expands a fundamental op from 2 lines (comment + code) to 7 (function def + call).